File: tt02/palindromo/Main.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

//Função que finaliza o programa
bool returnEnd(char fim []){
    bool result;

    result = (strlen(fim) == 4 && fim[0] == 'F' && fim[1] == 'I' && fim[2] == 'M');

    return result;
}

//Função que verifica se é um palíndromo ou não
int verifyPalidrome(char input[]){
    int j = strlen(input) - 2;
    int i = 0;
    int result = 1;

    while (i < j){
        if(input[i] != input[j])
            result = 0;
        i++;
        j--;
    }   

    return result;
}

int main(){
    char input[200];
    int output = 1;
    do{
        fflush(stdin);
        fgets(input, 200, stdin);
        output = verifyPalidrome(input);
        if(output == 1)
            printf("SIM\n");
        else
            printf("NAO\n");
    }while(returnEnd(input) == false);
    
    return 0;
}

File: tt02/palindromo/Main.java
import java.util.Scanner;
public class Main {
    
    //Função que finaliza o programa
    public static boolean returnEnd(String fim){
        boolean result;

        result = (fim.length() == 3 && fim.charAt(0) == 'F' && fim.charAt(1) == 'I' && fim.charAt(2) == 'M');

        return result;
    }

    //Função que verifica se é um palíndromo ou não
    public static String verifyPalidrome(String input) {
        int i = 0;
        int j = input.length() - 1;
        String result = "SIM";

        while (i < j){
            if(input.charAt(i) != input.charAt(j))
                result = "NAO";
            i++;
            j--;
        }   
        return result;
    }

    public static void main(String[] args) {
        String[] in = new String[1000];
        String[] out = new String[1000];
        int inputNumber = 0;
        Scanner keyboard = new Scanner(System.in);

        do{
            in [inputNumber] = keyboard.nextLine();
            out [inputNumber] = verifyPalidrome(in[inputNumber]);
            System.out.println(out[inputNumber]);
            
        } while (returnEnd(in[inputNumber++]) == false);
    }
}